libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx.h

```diff
@@ -0,0 +1,709 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___CUDA_PTX_H
+#define  _LIBCUDACXX___CUDA_PTX_H
+
+#include "../cstdint" // uint32_t
+#include "../../../../../../nv/target" // __CUDA_MINIMUM_ARCH__ and friends
+
+#include "cuda/std/detail/libcxx/include/__cuda/ptx/ptx_isa_target_macros.h"
```

Comment:
question: Thrust relies on libcu++ even when CUDA is not used. To my understanding, this is a supported use case. Should we guard PTX headers with something like the code below not to break this?
```c++
#ifdef __cuda_std__
```  
Otherwise, if this includes leak to other headers (currently only leaks to cuda::std::barrier), we might break more places. I'd defer the question to @jrhemstad and @miscco.

Reply:
This is an internal header that is only ever included through `<__cuda/barrier>` which asserts that we defined `__cuda_std__` 

I have also added that requirement to this header.

Reply:
Actually yes, if someone includes `<cuda/ptx>` thex explicitly want our ptx instructions. If this ever leaks I want to know it and not silently hide it through some macro

Reply:
Right, we just defer hiding ptx header to the user. I'd prefer to do this on our end, because it can benefit us in the context where we use new ptx, say, in atomic, and suddenly break host usage. It seems to be less effort and safer default to hide header once than not forgetting to hide it on every usage.

Reply:
The downside to hiding header is that it gives unclear errors to the user. Having an assertion provides readable error, which seems as good-enough reason to put more work on our end. Resolving this conversation.